### PR TITLE
kconfig: default to LEGACY_INCLUDE_PATH=n

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Migrate the remaining code to the "zephyr/" prefixed #include and flip the default.

Dependencies:

- [x] https://github.com/zephyrproject-rtos/mcuboot/pull/82
  - [x] https://github.com/mcu-tools/mcuboot/pull/1421
  - [x] https://github.com/zephyrproject-rtos/mcuboot/pull/83
- [x] https://github.com/zephyrproject-rtos/chre/pull/3
  - [x] https://android-review.googlesource.com/c/2160657
  - [x] https://github.com/zephyrproject-rtos/chre/pull/4
- [x] https://github.com/zephyrproject-rtos/sof/pull/17
- [x] https://github.com/zephyrproject-rtos/segger/pull/12
- [x] https://github.com/zephyrproject-rtos/hal_silabs/pull/18
- [x] https://github.com/zephyrproject-rtos/hal_nordic/pull/119

### CODE REVIEW

**Review by ppryga-nordic — DISMISSED** [review]

**Comment by gmarull** [comment]
@fabiobaltieri please rebase and update shas

**Comment by fabiobaltieri** [comment]
Cool, let's see what else broke since the last run...

**Comment by fabiobaltieri** [comment]
Had to pull back the segger module by one patch, HEAD there needs one more patch (#48513). Fixed few more includes and squashed them in the two initial commits.

**Review by gmarull — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
Here goes nothing... will need a rebase if https://github.com/zephyrproject-rtos/zephyr/pull/48513 goes in first.

File: Kconfig.zephyr [hunk 1]
```diff
@@ -845,7 +845,7 @@ endmenu
 
 config LEGACY_INCLUDE_PATH
 	bool "Allow for the legacy include paths (without the zephyr/ prefix)"
-	default y
+	default n
```
Comment (stephanosio) [inline Kconfig.zephyr:848]: `default n` is redundant and should be removed.

Comment (fabiobaltieri) [reply Kconfig.zephyr]: Right! Dropped.

**Comment by fabiobaltieri** [comment]
Dropped "default n" from the config option definition (it's the default).

**Review by gmarull — DISMISSED** [review]

**Review by stephanosio — DISMISSED** [review]

**Comment by stephanosio** [comment]
@fabiobaltieri #48513 went in first. this now needs a rebase.

**Comment by fabiobaltieri** [comment]
> @fabiobaltieri #48513 went in first. this now needs a rebase.

Cool, rebased, does not touch the segger module anymore since that's already done!

**Review by gmarull — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
Thanks folks! Let's do this! :-)

**Pull request merged**